Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift

```diff
@@ -237,6 +302,18 @@ public extension Sequence {
     func first<T: Equatable>(where keyPath: KeyPath<Element, T>, equals value: T) -> Element? {
         return first { $0[keyPath: keyPath] == value }
     }
+
+    /// SwifterSwift: Returns the first element of the sequence with having property by given key path equals to given
+    /// `value`.
+    ///
+    /// - Parameters:
+    ///   - map: Function for `Element` to compare.
+    ///   - value: The value to compare with `Element` property.
+    /// - Returns: The first element of the collection that has property by given map function equals to given `value` or
+    /// `nil` if there is no such element.
+    func first<T: Equatable>(where map: (Element) throws -> T, equals value: T) rethrows -> Element? {
```

Comment:
I love that you made this function throwable. Do you want to do that for the other new functions you created?

Reply:
Oh, for sure!